draft-ietf-quic-transport.md

```diff
@@ -1414,50 +1414,47 @@ Handshake[0]: CRYPTO[FIN], ACK[0]
-Destination Connection ID is used to determine packet protection keys for
-Initial packets.
+Initial or Retry packet from the server, the client populates the DCID field
+with an unpredictable value.  This DCID MUST be at least 8 bytes in length.
+Until a packet is received from the server, the client MUST use the same DCID
+value on all packets in this connection. This DCID is used to determine packet
+protection keys for Initial packets.
 
-The client populates the Source Connection ID field with a value of its choosing
-and sets the SCID Len field to indicate the length.
+The client populates the SCID field with a value of its choosing and sets the
+SCID Len field to indicate the length.
 
-The first flight of 0-RTT packets use the same Destination and Source Connection
-ID values as the client's first Initial.
+The first flight of 0-RTT packets use the same DCID and SCID values as the
+client's first Initial packet.
 
 Upon first receiving an Initial or Retry packet from the server, the client uses
-the Source Connection ID supplied by the server as the Destination Connection ID
-for subsequent packets, including any subsequent 0-RTT packets.  That means that
-a client might change the Destination Connection ID twice during connection
-establishment, once in response to a Retry and once in response to the first
-Initial packet from the server. Once a client has received an Initial packet
-from the server, it MUST discard any packet it receives with a different Source
-Connection ID.
-
-A client MUST only change the value it sends in the Destination Connection ID in
-response to the first packet of each type it receives from the server (Retry or
-Initial); a server MUST set its value based on the Initial packet.  Any
-additional changes are not permitted; if subsequent packets of those types
-include a different Source Connection ID, they MUST be discarded.  This avoids
-problems that might arise from stateless processing of multiple Initial packets
-producing different connection IDs.
-
-The connection ID can change over the lifetime of a connection, especially in
-response to connection migration ({{migration}}); see {{issue-cid}} for details.
+the SCID supplied by the server as the DCID for subsequent packets, including
+all subsequent 0-RTT packets. This means that a client might change the DCID
+twice during connection establishment, once in response to a Retry and once in
+response to an Initial packet from the server. Once a client has received an
+Initial packet from the server, it MUST discard any subsequent packet it
+receives with a different Source Connection ID.
+
+A client MUST change the DCID value it sends in response to only the first
+received Initial or Retry packet.  A server MUST set its DCID value based on the
+first received Initial packet. Any further changes to the DCID are not
+permitted; if subsequent Initial packets include a different SCID, they MUST be
```

Comment:
are only permitted if the values are taken from NEW_CONNECTION_ID frames; if subsequent